src/serialized-analysis.ts

```diff
@@ -0,0 +1,425 @@
+export interface SerializedAnalysis { packages: Package[]; }
+
+export interface Package {
+  /** The name of the package, like `paper-button` */
+  name: string;
+  /** The version, extracted from package.json, bower.json, etc as available. */
+  version: string;
+
+  /** The npm metadata of the package, if any. */
+  npmPackage?: NpmPackage;
+
+  /** The bower metadata of the package, if any. */
+  bowerMetadata?: BowerMetadata;
+
+  /** Elements found inside the package. */
+  elements: Element[];
+}
+
+export interface Element {
+  /**
+   * The path, relative to the base directory of the package.
+   *
+   * e.g. `paper-input.html` or `app-toolbar/app-toolbar.html` (given that
+   * app-toolbar lives in the app-layout package).
+   */
+  path: string;
+
+  /** The tagname that the element registers itself as. e.g. `paper-input` */
+  tagname: string;
+
+  /** A markdown description for the element. */
+  description: string;
+
+  /**
+   * Paths, relative to `this.path` to demo pages for the element.
```

Comment:
rather that "this file" (by which I meant the elements.json file) we can say package directory